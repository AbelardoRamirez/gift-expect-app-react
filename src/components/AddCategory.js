import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [newCategorie, setNewCategorie] = useState('');

    const handleClick = () => {
        if (newCategorie !== '' && newCategorie.length > 3) {
            setCategories(cat => [newCategorie, ...cat]);
            setNewCategorie('');
        } else {
            console.log('====================================');
            console.log(`La Categoria debe de contener >= 3 caracteres`);
            console.log('====================================');
        }
    }

    const handleNewCategorie = (e) => {
        setNewCategorie(e.target.value);
    }

    return (
        <div className="row border my-2 shadow-lg">
            <div className="col">
                <input type="text" className="form-control my-2" onChange={handleNewCategorie} value={newCategorie} />
                <button className="btn btn-block btn-dark my-2" onClick={handleClick}>
                    Agregar Nuevo Elemento
                </button>
            </div>
        </div>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;