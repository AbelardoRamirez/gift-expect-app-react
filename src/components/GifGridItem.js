import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ id, type, imagen, title, source, import_datetime }) => {

    return (
        <div key={id} className="col-3 animate__animated animate__fadeIn">
            <hr className="hr shadow-lg border" />
            <div className="card text-center">
                <div className="card-header">
                    T: <span className="font-weight-bolder">{type}</span> - ID: <span className="font-weight-lighter">{id}</span>
                </div>
                <div className="card-body">
                    <img src={imagen} alt={title} className="img-fluid" />
                    <h5 className="card-title">{title}</h5>
                    <p className="card-text">Source: <span className="font-weight-lighter text-monospace">{source}</span></p>
                </div>
                <div className="card-footer text-muted">
                    {import_datetime}
                </div>
            </div>
        </div>
    );
}

GifGridItem.propTypes = {
    id: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    imagen: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    source: PropTypes.string.isRequired,
    import_datetime: PropTypes.string.isRequired,
}

export default GifGridItem;