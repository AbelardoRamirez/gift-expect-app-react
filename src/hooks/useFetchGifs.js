import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/GetGifs'

export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        console.log(`Prueba de useFetchGifs`);
        getGifs(category).then(resp => {
            console.log(resp);
            setState({
                data: resp,
                loading: false,
            });
        })
    }, [category]);

    return state;
};
