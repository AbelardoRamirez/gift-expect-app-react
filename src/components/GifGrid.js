import React, { useEffect } from 'react';
//import { getGifs } from '../helpers/GetGifs';
import { useFetchGifs } from '../hooks/useFetchGifs'
import GifGridItem from './GifGridItem';



export const GifGrid = ({ category }) => {
    const { loading, data: imagenes } = useFetchGifs(category);

    useEffect(() => {

    }, [category]);

    return (
        <div className="row">
            {
                loading
                &&
                <div className="col animate__animated animate__repeat-2 animate__flash">
                    <h6 className="h6 font-weight-lighter text-muted">Loading Content Please Wait...</h6>
                </div>
            }
            {imagenes.map((img) => {
                return (<GifGridItem {...img} />)
            })}
        </div>
    )
}
