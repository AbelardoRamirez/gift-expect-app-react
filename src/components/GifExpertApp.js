import React, { useState } from 'react'
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div className="row justify-content-center">
            <div className="col">
                <h2 className="h2 font-weight-lighter text-center text-muted my-3">
                    GiftExpertApp
                </h2>
                <AddCategory setCategories={setCategories} />
                <hr className="hr shadow-lg" />
                <div className="row justify-content-center">
                    <div className="col">
                        {
                            categories.map(cat => (<GifGrid category={cat} key={cat} />))
                        }
                    </div>
                </div>
            </div>
        </div>
    )
}
export default GifExpertApp