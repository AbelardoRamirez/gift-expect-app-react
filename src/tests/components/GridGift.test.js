import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';
jest.mock('../../hooks/useFetchGifs');



describe('Pruebas en el Componente <GriGift/>', () => {

    test('Debe de mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });
        const wrapper = shallow(<GifGrid category="react" />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Mostrar Items cargados correctamente', () => {

        const gifs = [{
            type: 'Games', id: 999, title: 'IMG_21-10-2020', import_datetime: new Date().getDate().toString(), imagen: 'https://media.giphy.com/media/ZVik7pBtu9dNS/giphy.gif', source: 'GIPHY.com'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category='react' />);

        // expect( wrapper ).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});