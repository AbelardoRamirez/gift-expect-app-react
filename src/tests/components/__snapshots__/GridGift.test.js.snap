// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <GriGift/> Debe de mostrar el componente correctamente 1`] = `
<div
  className="row"
/>
`;
